activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -54,16 +54,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
```

Comment:
> I would like nothing more than to stop bikeshedding on this, but you guys are relentless

hhahahhaah true, let's finish this. If there is a way to test your previous comment about mutating the object lets make sure we have tests for that or we will remove the freeze later.